api/presenters/PostPresenter.js

```diff
@@ -138,9 +139,28 @@ const postListRelations = (userId, opts = {}) => {
   ])
 }
 
+const presentProjectActivity = function (post, data, userId, relationsOpts) {
+  if (post.type !== 'project') return post
+  return Post.query(q => {
+    q.where({parent_post_id: post.id})
+    q.orderBy('updated_at', 'desc')
+    q.limit(1)
+  })
+  .fetch({withRelated: postListRelations(userId, relationsOpts || {})})
+  .then(child => {
+    if (!child || Math.abs(post.updated_at.getTime() - child.get('updated_at').getTime()) > 1000) return post
+    child = postAttributes(child, userId, relationsOpts)
+    child.project = post
+    child.type = 'project-activity'
+    normalizePost(child, data, true)
```

Comment:
the third argument should only be set to `true` at the end of the whole iteration -- so you could either use the index from `map`, e.g.:

```js
Promise.map(data.posts, (p, i) => presentProjectActivity(post, data, userId, relationsOpts, i == data.posts.length - 1))
```

or you could just always omit the third argument here, and just call `uniqize` in `then` after Promise.map, like we do in SearchController.